### PR TITLE
ISPN-9370 Always consult CacheLoader if entry is not in data container

### PR DESCRIPTION
**Pull request by wburns**
* Add includeStores method to load method on PersistenceManager

https://issues.jboss.org/browse/ISPN-9370

### CODE REVIEW

**Comment by wburns** [comment]
This PR will most definitely conflict with https://github.com/infinispan/infinispan/pull/6131. If possible please merge the other PR first :)

**Comment by arkaev** [comment]
looks good, can't reproduce bug locally any more

**Comment by wburns** [comment]
Yeah that is what happens when you push something late and night and should be in bed, haha.

**Comment by wburns** [comment]
Fixed.

File: core/src/main/java/org/infinispan/persistence/PersistenceUtil.java [hunk 1]
```diff
@@ -129,100 +129,101 @@ public static InternalMetadata internalMetadata(InternalCacheValue icv) {
             isLoaded);
    }
 
-   public static <K, V> InternalCacheEntry<K,V> loadAndStoreInDataContainer(DataContainer<K, V> dataContainer, int segment,
```
Comment (danberindei) [inline core/src/main/java/org/infinispan/persistence/PersistenceUtil.java]: The package is `@public`, so I'm not sure about these API changes

Comment (wburns) [reply core/src/main/java/org/infinispan/persistence/PersistenceUtil.java]: I know the diff is really ugly, but the public methods are still there. If not then that is an issue.

File: core/src/test/java/org/infinispan/expiration/impl/CustomLoaderNonNullWithExpiration.java [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+import org.infinispan.util.TimeService;
+import org.testng.annotations.Test;
+
+/**
+ * Test to verify that a loader that always returns a non null value
+ * @author wburns
+ * @since 9.4
+ */
+@Test(groups = "functional", testName = "expiration.impl.CustomLoaderNonNullWithExpiration")
+public class CustomLoaderNonNullWithExpiration extends SingleCacheManagerTest {
```
[the remark this answers is not in the document]
Comment (wburns) [reply core/src/test/java/org/infinispan/expiration/impl/CustomLoaderNonNullWithExpiration.java]: Fixed

**Comment by wburns** [comment]
Updated, also reordered the `PersistenceUtil` methods to make the diff a little nicer.

**Pull request merged**